client/src/views/Lobby.vue

```diff
@@ -19,19 +19,20 @@ if (rawLobbyId && typeof rawLobbyId == "string") {
 }
 
 onBeforeRouteLeave(() => {
-    states.gameSocket?.send("leaveLobby")
+    if (states.lobby == lobbyId) {
+        states.gameSocket?.send("leaveLobby")
+    }
 })
 
 onMounted(() => {
     if (states.lobby != lobbyId) { // then we either joined or switched
-        states.gameSocket?.setOnMessage((message, sk) => {
+        states.gameSocket?.messageHandlers.set("addToLobby", (message, sk) => {
```

Comment:
What about removing this handler on unmount?

Reply:
Just made it that it will be removed on every pattern matching